windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs

```diff
@@ -233,8 +234,12 @@ public bool ToolbarSeparatorRight
         var transfersCache = store.Transfers.Connect().ObserveOnDispatcher()
             .Transform(m => new TransferViewModel(controller, m, locale)).DisposeMany()
             .Bind(out transfers).AsObservableCache().DisposeWith(subscriptions);
-        subscriptions.Add(transfersCache.Connect().Subscribe(OnTransfersChanged));
         subscriptions.Add(transfersCache.CountChanged.Subscribe(OnTransfersCountChanged));
+        subscriptions.Add(Observable.FromEvent<Transfer>(
+            h => controller.RevealTransfer += h,
+            h => controller.RevealTransfer -= h)
+            .Select(transfersCache.WatchValue).Switch()
```

Comment:
If transfersCache doesn't have Transfer cached, with Lookup would return `Optional.None`.
This guarantees that a valid value is passed on.